<launch>
    <arg name="nodelet_manager_on" default="true" />
    <arg name="nodelet_manager_name" default="rio_nodelet_manager" />

    <arg name="rio_on" default="true" />
    <arg name="rio_queue_size" default="1" />
    <arg name="rio_odom_frame_id" default="odom" />

    <arg name="visualize" default="false"/>

    <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" cwd="node" output="screen" if="$(arg nodelet_manager_on)"/>

    <group if="$(arg rio_on)">
        <node pkg="nodelet" type="nodelet" name="rio" args="load rio::RioNodelet $(arg nodelet_manager_name)" output="screen">
			<rosparam file="$(find rio)/cfg/rio.yaml" />
            <param name="queue_size" value="$(arg rio_queue_size)"/>
            <param name="odom_frame_id" value="$(arg rio_odom_frame_id)"/>
        </node>

        <!--node pkg="rio" type="rio_node" name="rio" output="screen" -->
			<!--rosparam file="$(find rio)/cfg/rio.yaml" />
            <param name="queue_size" value="$(arg rio_queue_size)"/>
            <param name="odom_frame_id" value="$(arg rio_odom_frame_id)"/>
        </node-->
        <node name="imu_to_radar_broadcaster" pkg="tf2_ros" type="static_transform_publisher" args="0.122 0.000 -0.025 0.67620958 0.67620958 -0.20673802 -0.20673802 '$(arg imu_frame_id)' '$(arg radar_frame_id)'" />
    </group>

    <node name="radar_to_cam_broadcaster" pkg="tf2_ros" type="static_transform_publisher" args="-0.020 -0.015 0.0 0.0 0.7071068 0.7071068 0.0 '$(arg radar_frame_id)' '$(arg cam_frame_id)'" />

    <group if="$(arg visualize)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rio)/cfg/rviz.rviz"/>
        <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot" args="--multiplot-config $(find rio)/cfg/rio_multiplot.xml --multiplot-run-all"/>
    </group>

</launch>
